### PR TITLE
Gracefully handle duplicate name with unique MD5

### PR DESCRIPTION
**Pull request by dbutenhof**
PBENCH-1112

In testing we're seeing duplicate tarball names with distinct MD5 values. We have reduced the importance of duplicate tarball names, relying on the dataset resource ID (MD5) for "uniqueness" internally. However the cache manager still maintains tarball files and MD5 companions on disk, and the upload API has to temporarily store the API payload somewhere. When the duplicate tarballs are uploaded concurrently, the `.open()` fails, leading to an internal server error.

Instead, move temporary upload files into an MD5-based subdirectory, and clean up that directory after upload completes.

__NOTE__: I'm pushing this early as a draft for testing and exposure. I still need to write some unit tests this afternoon before it's formally ready for review.

### CODE REVIEW

File: lib/pbench/server/api/resources/upload_api.py [hunk 1]
```diff
@@ -236,9 +238,24 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
...
+            # failed, so that's not a concern.)
+            try:
+                tmp_dir = self.temporary / md5sum
+                tmp_dir.mkdir()
+            except FileExistsError:
+                raise CleanupTime(
+                    HTTPStatus.CONFLICT,
+                    "Temporary upload directory already exists",
+                )
+            upload_dir = tmp_dir
```
Comment (webbnh) [inline lib/pbench/server/api/resources/upload_api.py:256]: Why have both `upload_dir` and `tmp_dir`?  It seems like either one would suffice.

**Review by webbnh — APPROVED** [review]
LGTM

File: lib/pbench/server/api/resources/upload_api.py (hunk 1 above)
Comment (dbutenhof) [reply lib/pbench/server/api/resources/upload_api.py:256]: It was a journey ... but you're right, they're now redundant and it does look a bit odd.

**Review by webbnh — APPROVED** [review]
👍

**Pull request merged**